docs/custom-productions.md

```diff
@@ -0,0 +1,81 @@
+# Custom productions
+
```

Comment:
This is still a bit vague (particularly the word "productions" ... don't need to change it, just needs an explanation). It might be good to expand this a little bit more to make it more clear when one might want to use this (i.e., maybe say it's good proprietary extensions).

Reply:
I borrowed the word from [Web IDL spec](https://heycam.github.io/webidl/#idl-grammar) which is again from [the CS field](https://en.wikipedia.org/wiki/Production_(computer_science)). (Although I'm not 1000% sure I'm using the word here correctly 😅

Anyway I'll try making it clearer.

Reply:
Ah, got it. I think it would be great to point to the Wikipedia article. That clarifies a lot!